src/big-int/bigint.cpp

```diff
@@ -466,6 +466,9 @@ BigInt &BigInt::operator=(BigInt const &y)
 
 BigInt &BigInt::operator=(char const *s)
 {
+  reallocate(small);
+  length = 0;
+  positive = true;
```

Comment:
Why is this needed?

Reply:
This intends to fix the problems from Issue #108

When reassigning from a string `scan_on` assumes that the `length` is already set. Specifics:

- `reallocate` to clean old data. This should fix replacing with a smaller number.
- `length` is set to 0 to start from the beginning to parse the number correctly
- `positive` is set to true because the `scan` function will only check if the number is negative with the "-" sign. So, if the older number was negative and the newer is positive BigInt will think that the number is negative.